cpp/include/cudf/copying.hpp

```diff
@@ -344,6 +344,12 @@ std::vector<gdf_column*> split(gdf_column const &          input_column,
                                gdf_index_type const*      indices,
                                gdf_size_type              num_indices);
 
+
+
+table shift(const table& in_table,
+            gdf_index_type period,
+            const gdf_scalar* fill_value);
```

Comment:
I'm uncomfortable with fill_value being a pointer. What is the need for allowing it to be `nullptr`?  It's easy enough for callers to pass a proper gdf_scalar. Is it just because you want to allow a default value? In that case, make it `const gdf_scalar fill_value = gdf_scalar{}`. The dtype will be GDF_invalid, but you can detect that and replace it with the output column's dtype.  Not perfect, but better than `nullptr` IMO.

Reply:
> Is it just because you want to allow a default value?
IMO, this function should not know a default value. I use nullptr because I don't want to use a valid at all, but only set the invalid bit to null. This prevents an extra copy, and leaves the underlying data as it is. I'm not sure if either of these are important, but `nullptr` seemed appropriate, given that the alternative would be an additional parameter (`bool set_value`), or a different sentinel value (aside from nullptr).

I can go about this differently. Under what circumstances do you think nullptr is appropriate?

Reply:
A few things are unclear to me in your response. Why shouldn't this function have a default fill value?  What does "set the invalid bit to null" mean? Do you mean "set the valid bit to null"? What extra copy are you preventing?

A `gdf_scalar` can be null/invalid, so a non-pointer parameter seems to fit the bill.